File: Matury CKE/2022 Czerwiec/PROGRAMOWANIE/python/zad4_4.py
# numbers=[i.strip() for i in open('../../DANE/przyklad.txt').readlines()]
numbers=[i.strip() for i in open('../../DANE/liczby.txt').readlines()]

checked=[]
count=[]

for i in range(len(numbers)):
    if numbers[i] not in checked:
        count.append([ numbers[i], numbers.count(numbers[i]) ])
        checked.append(numbers[i])

total_diff=len(count)
total_double=len([i for i in count if i[1]==2])
total_tripple=len([i for i in count if i[1]==3])

print(f'4.4. {total_diff} {total_double} {total_tripple}')

File: Matury CKE/2022 Czerwiec/Programowanie/C++/zad4_4.cpp
#include <iostream>
#include <string>
#include <fstream>
#include <vector>
#include <algorithm>

int main()
{
    std::cout << "4.4. ";

    // std::ifstream dane("C:/Matury-Informatyka/Matury CKE/2022 Czerwiec/DANE/przyklad.txt");
    std::ifstream dane("C:/Matury-Informatyka/Matury CKE/2022 Czerwiec/DANE/liczby.txt");

    std::vector<std::string> numsList;

    std::string num;
    while(dane >> num){
        numsList.push_back(num);
    }

    std::vector<std::string> diffNums;
    std::vector<std::string> twoTimes;
    std::vector<std::string> threeTimes;

    for(auto a: numsList){
        int timesIn = count(numsList.begin(), numsList.end(), a);

        bool foundInAll = find(diffNums.begin(), diffNums.end(), a) != diffNums.end();
        if(!foundInAll){
            diffNums.push_back(a);
        }

        switch (timesIn) {
            case 2:
                {
                    bool foundInTwo = find(twoTimes.begin(), twoTimes.end(), a) != twoTimes.end();
                    if(!foundInTwo){
                        twoTimes.push_back(a);
                    }
                }
                break;
            case 3:
                {
                    bool foundInThree = find(threeTimes.begin(), threeTimes.end(), a) != threeTimes.end();
                    if(!foundInThree){
                        threeTimes.push_back(a);
                    }
                }
                break;
            default:
                break;
        }
    }

    std::cout<<diffNums.size() << " " << twoTimes.size() << " " << threeTimes.size();

    return 0;
}
